### PR TITLE
Remove BaseDistribution from plugin api

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #8386

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/content/handler.py [hunk 1]
```diff
@@ -177,23 +175,21 @@ def _match_distribution(cls, path):
...
+                )
         else:
             try:
-                model_class = cls.distribution_model
                 return cls.distribution_model.objects.get(base_path__in=base_paths)
             except ObjectDoesNotExist:
                 log.debug(
-                    _("{model_name} not matched for {path} using: {base_paths}").format(
-                        model_name=model_class.__name__, path=path, base_paths=base_paths
+                    _("Distributoin not matched for {path} using: {base_paths}").format(
```
Comment (dralley) [inline pulpcore/content/handler.py:191]: typo

File: pulpcore/app/models/publication.py [hunk 2]
```diff
@@ -290,38 +287,9 @@ class BaseDistribution(MasterModel):
...
 
     def __init__(self, *args, **kwargs):
-        """Initialize a BaseDistribution and emit deprecation warnings"""
-        deprecation_logger.warn(
-            _(
-                "BaseDistribution is deprecated and could be removed as early as pulpcore==3.13; "
-                "use pulpcore.plugin.models.Distribution instead."
-            )
+        raise Exception(
+            "Using BaseDistribution is no longer supported. Please use Distribution instead"
```
Comment (mdellweg) [inline pulpcore/app/models/publication.py:291]: Does makemigration instantiate such models? Can you verify that it does not blow up here?
Is "NotImplementedError" a better matching exception?

Comment (ipanova) [reply pulpcore/app/models/publication.py:291]: please use `pulpcore.plugin.models.Distribution` instead

Comment (daviddavis) [reply pulpcore/app/models/publication.py:291]: > Does makemigration instantiate such models? Can you verify that it does not blow up here?

Good question. I tested and it seems to work.

> Is "NotImplementedError" a better matching exception?

I considered it but reading the description for NotImplementedError and it sounds like it's maybe not a good fit?

https://docs.python.org/3/library/exceptions.html#NotImplementedError

**Review by dralley — APPROVED** [review]

**Review by mdellweg — APPROVED** [review]
LGTM

**Pull request merged**